### PR TITLE
Optimize auto approval workflow

### PR DESCRIPTION
**Pull request by banesullivan**
I'm not totally sure if this will fix things, but I'm hoping that by moving the condtional from a workflow step to the job level, GitHub Actions will be able to parse whether or not it needs to spin up a runner or not.

At present, this workflow is eating up a ton of our CI bandwidth as it is spinning up a runner for every single PR comment *and* every time a comment is edited.

I went a step further and turned off runs for comment edits.

### CODE REVIEW

**Comment by tkoyama010** [comment]
ping @germa89

**Comment by banesullivan** [comment]
I may go spin this up in a dummy repo to test

**Comment by banesullivan** [comment]
Confirmed in a sperate repo that this should do it!

Screenshot 2023-12-11 at 10 52 25 PM

File: .github/workflows/auto-approve.yml [hunk 1]
```diff
@@ -2,25 +2,24 @@ name: Approve PRs
...
     permissions:
       pull-requests: write
     runs-on: ubuntu-latest
     steps:
       - uses: hmarr/auto-approve-action@v3
-        if: |
-          contains(github.event.comment.body, '@pyvista-bot LGTM') && contains(env.ALLOWED_AUTHORS, github.event.comment.user.login)
         with:
-          review-message: ":white_check_mark: Approving this PR because [${{ github.event.comment.user.login }}](https://github.com/${{ github.event.comment.user.login }}) said so in [here](${{ github.event.comment.html_url }}) :grimacing:"
+          review-message: ":white_check_mark: Approving this PR because [${{ github.event.comment.user.login }}](https://github.com/${{ github.event.comment.user.login }}) said so in [here](${{ github.event.comment.html_url }}) :shipit:"
```
Comment (tkoyama010) [inline .github/workflows/auto-approve.yml:23]: I didn't know about it.
[On GitHub, what is the significance of the "Ship It" squirrel?](https://www.quora.com/On-GitHub-what-is-the-significance-of-the-Ship-It-squirrel)

**Review by tkoyama010 — APPROVED** [review]
LGTM

File: .github/workflows/auto-approve.yml (hunk 1 above)
Comment (banesullivan) [reply .github/workflows/auto-approve.yml:23]: :shipit:

**Pull request merged**